// Copyright 2022-2024 ETSI SDG TeraFlowSDN (TFS) (https://tfs.etsi.org/)
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package interdomain;

import "context.proto";

service InterdomainService {
  rpc Authenticate              (context.TeraFlowController) returns (context.AuthenticationResult) {} // Deprecated
  rpc RequestSlice              (context.Slice             ) returns (context.SliceId             ) {}
  rpc LookUpSlice               (context.Slice             ) returns (context.SliceId             ) {}
  rpc OrderSliceFromCatalog     (context.Slice             ) returns (context.Slice               ) {}
  rpc CreateSliceAndAddToCatalog(context.Slice             ) returns (context.Slice               ) {}
  rpc OrderSliceWithSLA         (context.Slice) returns (context.SliceId) {} // If slice with SLA already exists, returns slice. If not, it creates it.
  rpc UpdateSlice               (context.Slice             ) returns (context.Slice               ) {}
  rpc DeleteSlice               (context.SliceId           ) returns (context.Empty               ) {}
}
